src/fit-curve.js

```diff
@@ -26,15 +26,16 @@ function fitCurve(points, maxError, progressCallback) {
         throw new TypeError("First argument should be an array");
     }
     points.forEach((point) => {
-        if(!Array.isArray(point) || point.length !== 2
-        || typeof point[0] !== 'number' || typeof point[1] !== 'number'){
-            throw Error("Each point should be an array of two numbers")
+        if(!Array.isArray(point) || point.some(item => typeof item !== 'number')
+        || point.length !== points[0].length) {
+            throw Error("Each point should be an array of numbers. Each point should have the same amount of numbers.");
         }
     });
+
     // Remove duplicate points
-    points = points.filter((point, i) =>
-        i === 0 || !(point[0] === points[i-1][0] && point[1] === points[i-1][1])
-    );
+    points = points.filter((point, i) => {
+      return i === 0 || !point.every((val, j) => val === points[i-1][j])

```

Comment:
why bringing `return` and curly braces here?

Reply:
Ah, a minute too late. Just pushed the commit to remove curly braces and return in this case.

Reply:
You made a great job! Let me fix a bit those last things and release ;-)

Reply:
ok. thanks! Go ahead. :)
I'll remember to use extra feature branches instead changing the master of a fork for future PRs, it certainly makes more sense.